### ISSUE TITLE
Clang-Tidy: modernize-use-nullptr does not make sense with spaceship operator

### ISSUE THREAD

**Issue by pavelkryukov** [llvm/llvm-project#53961]
Reopening #45580, as the problem was not solved:

https://godbolt.org/z/3hed7vPvM

```c++
#include <compare>

struct A {
  int a;
  constexpr auto operator<=>(const A&) const = default;
};

bool foo(A x, A y) { return x < y; }
```

Command line:
```bash
clang-tidy-13 -checks='hicpp-use-nullptr' tidy20.cpp -- -std=c++20
```

Output:
```
<source>:5:56: warning: use nullptr [hicpp-use-nullptr]
    constexpr auto operator<=>(const A&) const = default;
                                                       ^
                                                        nullptr
```

Also, for strange reasons I cannot escape it with NOLINTNEXTLINE with local clang-tidy-13.
Godbolt's trunk version suppresses warning correctly.

**Comment by JonasToth** [comment]
for the suppression: maybe this check is aliased, did you use `NOLINTNEXTLINE(modernize-use-nullptr)`?

**Comment by pavelkryukov** [comment]
Yes, I did.

**Comment by jensmassberg** [comment]
`modernize-use-nullptr` gives a warning for

```
#include <compare>

struct A {
  int a;
  constexpr auto operator<=>(const A&) const = default;
};

bool foo(A x, A y) { return x < y; }
```

but it doesn't warn for

```
#include <compare>

struct A {
  constexpr auto operator<=>(const A&) const = default;
};

bool foo(A x, A y) { return x < y; }
```

nor for

```
namespace std {
struct strong_ordering {
  int n;
  constexpr operator int() const { return n; }
  static const strong_ordering equal, greater, less;
};
constexpr strong_ordering strong_ordering::equal = {0};
constexpr strong_ordering strong_ordering::greater = {1};
constexpr strong_ordering strong_ordering::less = {-1};
} // namespace std

struct A {
  int a;
  constexpr auto operator<=>(const A&) const = default;
};

bool foo(A x, A y) { return x < y; }
```

The latter one is equivalent to the C++20 test for `modernize-use-nullptr` .
